enhancements/insights/container-scanner.md

```diff
@@ -0,0 +1,532 @@
+
+The container scanner functionality is provided by an image that must be pullable from the `openshift-insights` namespace and present in the cluster at installation.
+
+#### Variation and form factor considerations [optional]
+
+This enhancement to the Insight Operator is not impacting the Control Plane of OpenShift and should be available regardless of the  form of OpenShift (Standalone OCP,  Microshift, Hypershift).
+
+### API Extensions
+
+This proposal does not require any API Extensions to the OpenShift Insights Operator.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+#### Hypershift [optional]
+
+This proposal only aims to gather data about the Data plane and will not scan any containers running in the Control Plane.
+
+#### Standalone Clusters
+
+This proposal is relevant for standalone clusters.
+
+#### Single-node Deployments or MicroShift
+
+This proposal affects single-node deployment as there is additional resource consumption
+when the Insights Operator gathers workload data that will affect the only worker node.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Container Scanner API
+
+The Insights Operator communicates with the Container Scanner pods deployed by its DaemonSet with a simple executable `/scan-containers`:
+
+* `/scan-containers`
+  * Scan the all the containers running on the worker node. The executable returns a JSON payload with the extracted data organized in a tree of `namespace/pod-name/container-id/runtime-info`:
+
+```json
+{
+  "my-namespace": {
+    "my-pod-1": {
+      "cri-o://xxxx": {
+        "os-release-id": "rhel",
+        "os-release-version-id": "8.8",
+        "runtime-kind": "Java",
+        "runtime-kind-version": "17.0.8",
+        "runtime-kind-implementer": "Red Hat, Inc.",
+        "runtimes": [
+          {
+            "name": "Quarkus",
+            "version": "3.4.1",
+          }
+        ],
+      },
+      "cri-o://yyyy": {
+        "os-release-id": "rhel",
+        "os-release-version-id": "9.2",
+      }
+    },
+    "my-pod-2": {
+      [...]
+    }
+  },
+  "other-namespace": {
+    "other-pod-1": {
+      "cri-o://zzzz": {
+        [...]
+      }
+    }
+  }
+}
+```
+
+#### Container Scanner Implementation
+
+Scanning a container is achieved by running on the same worker node that the scanned container and looking at the host process table. The container scanner enters the container’s process namespaces to extract meaningful information about the runtime stack.
+
+The extraction of data from the running container is achieved by executing “fingerprint” executables in the process `mnt` namespace to find the relevant information by reading specific files. These fingerprints executables are self-contained executables
```

Comment:
> ... find the relevant information by reading specific files

maybe it's worth saying which files?

Reply:
that's dependent on the detected runtimes and is out of scope of the integration. However, as an example, I mentioned the `/etc/os-release` file as the source of info for the `Os` and `OsVersion` fields.